Automattic/wp-calypso
# chore: upgrade `react-query` (v3) to `@tanstack/react-query` (v4)

## Proposed Changes

* Upgrade the `react-query@3` package to `@tanstack/react-query@4`
* Include any adjustments necessary because of breaking changes

## Testing Instructions

[Overview of changes between v3 and v4.](https://tanstack.com/query/latest/docs/react/guides/migrating-to-react-query-4)

**Please note**, this PR touches almost all queries and mutations for which we use the `react-query` library (now `@tanstack/react-query`) as well as a few other areas in the Calypso codebase (except for a few `/apps` directories which are not imported or don't import related code from Calypso or other packages.

* Verify Calypso builds well and does not contain any new type errors or warnings
* Depending on the query or mutation, verify data is fetched as expected (compare with prod) and no errors are thrown during runtime
* Verify the local persistence plugin still works as expected

## Pre-merge Checklist
